cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -416,19 +416,12 @@ private void doPerformAction(FullHttpRequest request, HttpResponder responder, S
     ProgramType programType = getProgramType(type);
     ProgramId program = applicationId.program(programType, programId);
     Map<String, String> args = decodeArguments(request);
-    String latestVersion = getLatestAppVersion(new NamespaceId(namespaceId), appId);
     // we have already validated that the action is valid
     switch (action.toLowerCase()) {
       case "start":
-        if (!appVersion.equals(latestVersion)) {
```

Comment:
why are we removing this?

Reply:
because we don't want to query the latest app version in handler. This constraint is now moved to ProgramLCservice.